packages/scene-composer/src/components/panels/scene-components/DataOverlayComponentEditor.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useCallback, useContext } from 'react';
+import { FormField, SpaceBetween, Textarea } from '@awsui/components-react';
+import { useIntl } from 'react-intl';
+
+import { IComponentEditorProps } from '../ComponentEditor';
+import { IDataOverlayComponentInternal, ISceneComponentInternal, useStore } from '../../../store';
+import { sceneComposerIdContext } from '../../../common/sceneComposerIdContext';
+import { Component } from '../../../models/SceneModels';
+
+import { DataBindingMapEditor } from './data-overlay/DataBindingMapEditor';
+
+export interface IDataOverlayComponentEditorProps extends IComponentEditorProps {}
+
+export const DataOverlayComponentEditor: React.FC<IDataOverlayComponentEditorProps> = ({
+  node,
+  component,
+}: IDataOverlayComponentEditorProps) => {
+  const sceneComposerId = useContext(sceneComposerIdContext);
+  const updateComponentInternal = useStore(sceneComposerId)((state) => state.updateComponentInternal);
+  const valueDataBindingProvider = useStore(sceneComposerId)(
+    (state) => state.getEditorConfig().valueDataBindingProvider,
+  );
+  const dataOverlayComponent = component as IDataOverlayComponentInternal;
```

Comment:
You should define this type in the props interface, so you don't need to cast it here.

Reply:
we will need to either cast here, or cast in `ComponentEditor` when passing into this component. it seems better to cast here so the component editor don't need to import all those interfaces

Reply:
I'm not sure I follow. This components signature isn't enforcing the interface, we're just casting it internally. Is there a reason this isn't on the props?

Reply:
I mean in [componentEditor](https://github.com/awslabs/iot-app-kit/blob/main/packages/scene-composer/src/components/panels/ComponentEditor.tsx#L40) when we pass `component` into the editor, the type of `component` is generic, so we either cast here, or cast before passing into the editor